### DISCUSSION TITLE
Rectangles not drawing to correct dimensions

### DISCUSSION THREAD

**Discussion by gingerbeardman** [Moddable-OpenSource/moddable#936] [Q&A] [answered]
## Problem

- I'm trying to block out an interface using rectangle to get the feel of how it might be to use.
- I'm drawing two filled rectangles in black, and two slightly smaller rectangles in white.

## Question
But I cannot explain why the black rectangles draw to the bottom of the screen?

And do let me know if there's an easier way of drawing non-filled rectangles!

## Screenshots

My goal (ignore the yellow, it's just for visibility):
![pd-on-m5](https://user-images.githubusercontent.com/49612/187746572-bc26b56f-e630-48c6-b136-dc46fd7ae1ad.png)

My current output:
![Screen shot 2022-08-31 at 19 00 58](https://user-images.githubusercontent.com/49612/187747738-91c9ea2b-9709-466f-95f1-d9590154029c.png)

My code:
```js
import Poco from "commodetto/Poco";
import {Outline} from "commodetto/outline";

let poco = new Poco(screen);
let white = poco.makeColor(255, 255, 255);
let black = poco.makeColor(0, 0, 0);

// clear screen
poco.begin();
	poco.fillRectangle(white, 0, 0, poco.width, poco.height);
poco.end();

let ox = 130;
let oy = 30;
let pdw = 800;
let pdh = 480;
let kl = 2;

// draw interface
poco.begin();
	poco.fillRectangle(black, 0, oy, ox-oy, pdh);
	poco.fillRectangle(white, kl, oy+kl, ox-oy-kl*2, pdh-kl*2);

	poco.fillRectangle(black, ox, oy, pdw, pdh);
	poco.fillRectangle(white, ox+kl, oy+kl, pdw-kl*2, pdh-kl*2);
poco.end();
```

**Comment by phoddie** [comment, marked as answer]
You need to draw every pixel in the area specified by `poco.begin`. In your code above ("draw interface"), that's the full screen. This is covered in a note in the `begin` [documentation](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/commodetto/poco.md#pocodrawingbegin).

**Reply by gingerbeardman** [reply]
Aha! Got it, I had not given that text enough attention and missed that note. Cheers!
